baseapp/helpers.go

```diff
@@ -3,19 +3,42 @@ package baseapp
 import (
 	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
 
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/tx"
 )
 
+// txEncoder creates a proto TxEncoder for testing purposes.
+func txEncoder(interfaceRegistry codectypes.InterfaceRegistry) sdk.TxEncoder {
+	marshaler := codec.NewProtoCodec(interfaceRegistry)
+	txCfg := tx.NewTxConfig(marshaler, tx.DefaultSignModes)
+
+	return txCfg.TxEncoder()
+}
+
 func (app *BaseApp) Check(tx sdk.Tx) (sdk.GasInfo, *sdk.Result, error) {
-	return app.runTx(runTxModeCheck, nil, tx)
+	// runTx expects tx bytes as argument, so we encode the tx argument into
+	// bytes. Note that runTx will actually decode those bytes again. But since
+	// this helper is only used in tests/simulation, it's fine.
+	bz, err := txEncoder(app.interfaceRegistry)(tx)
+	if err != nil {
+		return sdk.GasInfo{}, nil, err
+	}
+	return app.runTx(runTxModeCheck, bz)
```

Comment:
I would suggest to put all these functions inside this file into a `baseapp/testutil` package (in a follow-up PR).